Periodic checking for sensor

JHahn96 opened the pull request:
loop change: try to initialize the sensor every second.
added new bool that is changed to true if "init()" failed
Before only tried once and stayed in an endless loop.

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,17 +9,18 @@ void setup() {
-  int ret = sdp.init();
-  if (ret == 0) {
+  bool error = false;
+  do{
```
The project style dictates a single space before curly braces `{`. Could you please add that?

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,17 +9,18 @@ void setup() {
       Serial.print("init(): success\n");
-  } else {
+    } else {
+	  error = true;
```
This line mixes tabs and spaces in the indentation which is against the project style. Please use exactly 2 spaces for each indentation level.

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,17 +9,18 @@ void setup() {
-
-  int ret = sdp.init();
-  if (ret == 0) {
+  bool error = false;
```
The only purpose of `error` seems to be to keep the loop going. In this case it might make more sense to have an infinite loop which exits (`break;`) on success - that doesn't require an additional variable.

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,17 +9,18 @@ void setup() {
+      delay(1000);
       }
-  }
+  }while(error);
```
space before `while` - also if we drop `error` the condition becomes `while(true)`

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,17 +9,18 @@ void setup() {
-      while (true) {
-        delay(1000);
+      delay(1000);
       }
```
Seems misaligned

abrauchli commented on examples/sdp3x_manual_address/sdp3x_manual_address.ino:
```diff
@@ -10,35 +10,37 @@ SDP3XSensor sdp(SDP3XSensor::SDP3X_I2C_ADDR_21); // Address 0x21
-      Serial.print("Differential pressure: ");
-      Serial.print(sdp.getDifferentialPressure());
-      Serial.print("Pa | ");
+    Serial.print("Differential pressure: ");
```
Although this fixes an inconsistency in the style, it's not a change that belongs in this commit i.e. it has nothing to do with "periodic checking". Please create a separate commit for this style fix. Also, it should be consistently applied to every example file, not just this one.

abrauchli reviewed (changes requested):
Thanks for the PR, please note the changes required in the comments. Also the commit message should be written in present tense and describe the "diff", maybe something like

Try to initialize the sensor every second *on failure*.
Prior to the change, only one attempt was made.

abrauchli commented on examples/sdp3x_default_address/sdp3x_default_address.ino:
```diff
@@ -9,33 +9,32 @@ void setup() {
+  do {
+    int ret = sdp.init();
+    if (ret == 0) {
+    Serial.print("init(): success\n");
```
The content of the `if` block is not indented properly. Please shift accordingly.

abrauchli reviewed (changes requested):
Please also check the comment on the style changes that should be split into their own commit.

JHahn96 committed:
Periodic checking for sensor

* loop change: try to initialize the sensor every second
* break if init does not fail, if it fails try again in endless loop
  (first solution with adding another bool, but this solution seems
  bether because there are less declarations)
* before only tried once and stayed in an endless loop.
* correcting code shift (code was not shifted according to the used
  style)

abrauchli commented on examples/sdp8xx/sdp8xx.ino:
```diff
@@ -1,3 +1,6 @@
+
+#include <sdpsensor.h>
```
this was probably added by mistake since it's already included

abrauchli commented on examples/sdp3x_manual_address/sdp3x_manual_address.ino:
```diff
@@ -1,7 +1,7 @@
-
 #include <sdpsensor.h>
 
+#include <Wire.h>
```
Is there a specific reason to change the include order here? Otherwise I'd leave as much as possible unchanged.

abrauchli commented:
ok, the changes are good. You should now rebase on top of master and submit one more time. We can do it together if you like, or you can try by yourself.
